packages/core/src/new-api/types/artifact.ts

```diff
@@ -12,3 +12,12 @@ export interface Artifact {
     Record<string, Array<{ length: number; start: number }>>
   >;
 }
+
+/**
+ * Retrieve artifacts based on contract name.
+ *
+ * @beta
+ */
+export interface ArtifactResolver {
+  load(contractName: string): Promise<Artifact>;
```

Comment:
I mentioned this somewhere else, but we need two different objects to resolve artifacts.

One is public, which does this, plus resolving `build-info`s. I would copy these signatures from HH:
* https://github.com/NomicFoundation/hardhat/blob/main/packages/hardhat-core/src/types/artifacts.ts#L13
* https://github.com/NomicFoundation/hardhat/blob/main/packages/hardhat-core/src/types/artifacts.ts#L42

The other one is internal, and reads/writes artifacts from the deployment directory.

Reply:
As mentioned above this is coming in as part of #109